enterprise/models.py

```diff
@@ -763,9 +772,39 @@ def unenroll(self, course_run_id):
         return False
 
     def update_session(self, request):
-        """Update the session of a request for this learner."""
+        """
+        Update the session of a request for this learner.
+        """
         request.session['enterprise_customer'] = self.enterprise_customer.serialized
 
+    def create_order_for_enrollment(self, course_run_id):
+        """
+        Create an order on the Ecommerce side for tracking the course enrollment of a enterprise customer user.
+        """
+        LOGGER.info("Creating order for enterprise customer with id [%s] for enrollment in course with id: [%s]",
+                    self.user_id,
+                    course_run_id)
+        # get instance of the EcommerceApiClient and attempt to create order
+        ecommerce_service_worker = get_ecommerce_worker_user()
+        if ecommerce_service_worker is not None:
+            try:
+                ecommerce_api_client = EcommerceApiClient(ecommerce_service_worker)
+                # pull data needed for creating the order
+                course_enrollments = [{
+                    'lms_user_id': self.user_id,
+                    'username': self.username,
+                    'email': self.user_email,
+                    'course_run_key': course_run_id,
+                }]
+                ecommerce_api_client.create_manual_enrollment_orders(course_enrollments)
+            except NotConnectedToOpenEdX:
```

Comment:
According to the documentation, it doesn't seem like this exception is likely to be occur, but I added it since the init call _can_ raise this error. Let me know if I'm being too cautious.

Reply:
The caution is good. A couple python tips:
- You can grab a raised exception, stringify it, and log it directly. This lets us re-use some error code strings. So instead of 
    ```
    except NotConnectedToOpenEdX:
    ```
    you could set something like
    ```
    failure_log_msg = "Could not create order for user with id [%s] for enrollment in course with id [%s]. Reason: [%s]"
    ```
    above this block, then do
    ```
    except NotConnectedToOpenEdX as exc:
        LOGGER.exception(failure_log_msg, user_id, course_id, str(exc))
    ```
- The above would also allow you to do something like this at the bottom:
```
else:
    LOGGER.warning(failure_log_msg, user_id, course_id, "Failed to retrieve a valid ecommerce worker."
```

- You generally want to only do the 1 thing that might cause the exception inside the `try` block. Python's `try` system has a couple options of block you can use:
`try`: code to run that might contain exception
`except`: catch the exception if it happens
`else`: only run if `try` ran without an exception
`finally`: run always after the top blocks are done (used for cleanup in some projects, not heavily used here)
So this could could be transformed into:
```
try:
    # create client
except:
    # log error
else:
    # make api call with data
```